### PR TITLE
feat: `avm/res/network/service-endpoint-policy`

### LINKED ISSUE #515: [Module Proposal]: 	`avm/res/network/service-endpoint-policy`
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes; or I'm proposing the module to be migrated from CARML/TFVM.

### Bicep or Terraform?

Bicep

### Module Classification?

Resource Module

### Module Name

avm/res/network/service-endpoint-policy

### Module Details

Migrate from CARML

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username (handle)

jeetgarg

### (Optional) Secondary Module Owner's GitHub Username (handle)

_No response_

### PR DESCRIPTION
**Pull request by jeetgarg**
## Description

Migration from CARML : avm.res.network.service-endpoint-policy
Closes: [Azure/AVM#515](https://github.com/Azure/Azure-Verified-Modules/issues/515#issuecomment-1972306114)

## Pipeline Reference

[avm.res.network.service-endpoint-policy](https://github.com/jeetgarg/bicep-registry-modules/actions/workflows/avm.res.network.service-endpoint-policy.yml)

Tested in my local environment done.

### CODE REVIEW

File: .github/workflows/avm.res.network.service-endpoint-policy.yml
Comment (AlexanderSehr) [inline .github/workflows/avm.res.network.service-endpoint-policy.yml:1]: Please remove comments like the above :)

File: .github/workflows/avm.res.network.service-endpoint-policy.yml
Comment (AlexanderSehr) [inline .github/workflows/avm.res.network.service-endpoint-policy.yml:85]:
```suggestion
  call-workflow-passing-data:
    name: "Run"
    permissions:
      id-token: write # For OIDC
      contents: write # For release tags
    needs:
      - job_initialize_pipeline
    uses: ./.github/workflows/avm.template.module.yml
    with:
      workflowInput: "${{ needs.job_initialize_pipeline.outputs.workflowInput }}"
      moduleTestFilePaths: "${{ needs.job_initialize_pipeline.outputs.moduleTestFilePaths }}"
      psRuleModuleTestFilePaths: "${{ needs.job_initialize_pipeline.outputs.psRuleModuleTestFilePaths }}"
      modulePath: "${{ needs.job_initialize_pipeline.outputs.modulePath}}"
    secrets: inherit
```

File: .vscode/launch.json
Comment (AlexanderSehr) [inline .vscode/launch.json:1]: An intended file?

File: .vscode/settings.json
Comment (AlexanderSehr) [inline .vscode/settings.json:17]: Please undo. Those settings seem to be very much local ;)

File: avm/res/network/service-endpoint-policy/main.bicep
Comment (AlexanderSehr) [inline avm/res/network/service-endpoint-policy/main.bicep:173]: These seem to be leftovers, yet the outputs should be added.

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (AlexanderSehr) [inline avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:139]: Where thse changes intended? It appears a local format setting was applied to the file

File: .github/workflows/avm.res.network.service-endpoint-policy.yml
Comment (jeetgarg) [reply .github/workflows/avm.res.network.service-endpoint-policy.yml]: Done.

File: .github/workflows/avm.res.network.service-endpoint-policy.yml
Comment (jeetgarg) [reply .github/workflows/avm.res.network.service-endpoint-policy.yml]: Done

File: .vscode/launch.json
Comment (jeetgarg) [reply .vscode/launch.json:1]: Please Ignore this file :)

File: .vscode/settings.json
Comment (jeetgarg) [reply .vscode/settings.json:17]: Done

File: avm/res/network/service-endpoint-policy/main.bicep
Comment (jeetgarg) [reply avm/res/network/service-endpoint-policy/main.bicep]: Done

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (jeetgarg) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:139]: I did not change or add anything to this file. I am not sure about this piece of code.

File: .vscode/launch.json
Comment (AlexanderSehr) [reply .vscode/launch.json:1]: Ignore is relative as it's very much there :D  If we keep it it will be added for everybody. Is there an advantage of having it in this repository? :)

File: .vscode/settings.json
Comment (AlexanderSehr) [reply .vscode/settings.json:17]: You may have missed to push your changes @jeetgarg  ;)

File: .vscode/launch.json
Comment (jeetgarg) [reply .vscode/launch.json:1]: no, what I meant from ignore was to not merge it. :)

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (jeetgarg) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:139]: It seems that this is similar to what is there in the repo:
image

File: .vscode/settings.json
Comment (AlexanderSehr) [inline .vscode/settings.json:15]: Intended removal?

File: .vscode/launch.json
Comment (AlexanderSehr) [reply .vscode/launch.json:1]: Fair, but I don't think we can cherrypick the changes. Either it's part of a PR or not. To not loose your changes (as you're working in your `main` branch) I suggest you create a branch from your main branch and re-open the PR from there. Then you can freely undo changes there without loosing anything.

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (AlexanderSehr) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:139]: Hey @jeetgarg, it absolutely is. It's just the autoformatter that left an impression in your case ;) In general, it's not much of an issue - but if we were to merge it, GitHub would detect it as a change and re-run the workflows of all modules as it's a platform script. Unfortunately GitHub triggers are just not that smart 😄

Comment (AlexanderSehr) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:139]: On our end we plan to implement custom vscode settings eventually to make sure this can't happen, but until now we unfortunately didn't have capacity to look into this.

Comment (AlexanderSehr) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:139]: Feel free to ping me internally if you need help

Comment (jeetgarg) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1]: Check now. If its still not right, i will ping you

File: .vscode/settings.json
Comment (jeetgarg) [reply .vscode/settings.json]: Corrected

File: .vscode/launch.json
Comment (jeetgarg) [reply .vscode/launch.json:1]: Removed the file. Check now

Comment (AlexanderSehr) [reply .vscode/launch.json:1]: looks good, thanks @jeetgarg . 
On that note - if you have a couple settings that you'd like to have as part of the `settings.json` file, please let us know in an issue :)

File: .vscode/settings.json
Comment (AlexanderSehr) [reply .vscode/settings.json]: Looks good, thank you sir 💪

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (AlexanderSehr) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1]: One line got corrected, the other 95 not 😄 But progress. 
All the other fixes worked though.

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (AlexanderSehr) [inline avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:1]: Last one, then we can merge. Please execute the following steps
- Go to your fork's settings, specifically the actions settings
- Disable all actions
- Go back to your fork's main page
- Sync latest upstream
- Enable actions again

Fin.

What this will do is get rid of the below change as it's only shown as you overwrote the file with the latest in upstream manually (which is good), but as your fork doesn't know about the latest upstream it thinks you implemented the below changes ;)

Comment (jeetgarg) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:1]: Did it. Check now.

**Review by AlexanderSehr — DISMISSED** [review]

File: avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1
Comment (AlexanderSehr) [reply avm/utilities/pipelines/staticValidation/compliance/module.tests.ps1:1]: Nope, off we go. Thanks @jeetgarg. Thank you for your patience

File: .github/CODEOWNERS
Comment (AlexanderSehr) [inline .github/CODEOWNERS:108]: Ah wait, overlooked this. The team either is not existing, was not added to the hierachy of teams yet, or wasn't confirmed in set hierachy.

Comment (AlexanderSehr) [reply .github/CODEOWNERS:108]: The requirements are described [here](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only), yet it comes down to creating both an owner & contributor team as per the shown naming convention and adding them to the correct 'parent' team so that the permissions in the BRM repository are configured correctly.

Comment (jeetgarg) [reply .github/CODEOWNERS]: Check now

Comment (AlexanderSehr) [reply .github/CODEOWNERS]: @krbar could you provide some guidance on this one? Other than me, you have contributed a gazillion of modules and know the process inside out :D

Comment (krbar) [reply .github/CODEOWNERS]: I think your teams are missing a parent team. The correct parents are:
- parent of the owners team: `avm-technical-reviewers-bicep`
- parent of the contributors team: `avm-module-contributors-bicep`

You can set the parent team via team settings. Please keep in mind that the parentship must be approved by someone from the parent team and once approved, you will see an indication of the team path (see below example for the bastion host team):

image

See also the specs for more detailed info: https://azure.github.io/Azure-Verified-Modules/specs/shared/#team-memberships

Comment (jeetgarg) [reply .github/CODEOWNERS]: I have done all of this. Now, waiting for the approval from the parent team.

File: .github/CODEOWNERS
Comment (AlexanderSehr) [inline .github/CODEOWNERS:108]:
```suggestion
/avm/res/network/service-endpoint-policy/ @Azure/avm-res-network-serviceendpointpolicy-module-owners-bicep @Azure/avm-core-team-technical-bicep 
```

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**
